src/main/java/com/frc/codex/indexer/impl/UploadIndexerImpl.java

```diff
@@ -0,0 +1,187 @@
+		ResponseInputStream<GetObjectResponse> responseInputStream = s3Client.getObject(getObjectRequest);
+		return new BufferedReader(new InputStreamReader(responseInputStream));
+	}
+
+	public String getStatus() {
+		return String.format("""
+						Upload Indexer:
+						\tFilings indexed this session: %s
+						\tUploads completed this session: %s""",
+				sessionFilingCount,
+				sessionUploadCount
+		);
+	}
+
+	private List<String> getUploadKeys() {
+		ListObjectsRequest listObjectsRequest = ListObjectsRequest.builder()
+				.bucket(properties.s3IndexerUploadsBucketName())
+				.build();
+		ListObjectsResponse response = s3Client.listObjects(listObjectsRequest);
+		return response.contents().stream()
+				.map(S3Object::key)
+				.filter(key -> key.toLowerCase().endsWith(".csv"))
+				.toList();
+	}
+
+	public boolean isHealthy() {
+		return true;
+	}
+
+	private void indexCsvRow(List<String> row) {
+		int column = 0;
+		RegistryCode registryCode;
+		if (row.size() == 8) {
+			// registry_code,download_url,company_name,company_number,external_filing_id,external_view_url,filing_date,document_date
+			registryCode = RegistryCode.valueOf(row.get(column++));
+		} else if (row.size() == 7) {
+			// download_url,company_name,company_number,external_filing_id,external_view_url,filing_date,document_date
+			registryCode = RegistryCode.COMPANIES_HOUSE;
+		} else {
+			throw new RuntimeException("Invalid CSV row: " + row);
+		}
+		String downloadUrl = row.get(column++);
+		String companyName = row.get(column++);
+		String companyNumber = row.get(column++);
+		String externalFilingId = row.get(column++);
+		if (this.databaseManager.filingExists(registryCode.getCode(), externalFilingId)) {
+			LOG.debug("Skipping existing filing: {}", externalFilingId);
+			return;
+		}
+		String externalViewUrl = row.get(column++);
+		LocalDateTime filingDate = parseDate(row.get(column++));
+		LocalDateTime documentDate = parseDate(row.get(column++));
+		UUID filingId = this.databaseManager.createFiling(NewFilingRequest.builder()
+				.companyName(companyName)
+				.companyNumber(companyNumber)
+				.documentDate(documentDate)
+				.downloadUrl(downloadUrl)
+				.externalFilingId(externalFilingId)
+				.externalViewUrl(externalViewUrl)
+				.filingDate(filingDate)
+				.registryCode(registryCode.getCode())
+				.build()
+		);
+		LOG.info("Indexed filing filing from CSV: ({}, {}) {}", registryCode, externalFilingId, filingId);
+	}
+
+	private List<String> parseCsvRow(String line) {
+		List<String> result = new ArrayList<>();
+		StringBuilder currentField = new StringBuilder();
+		boolean inQuotes = false;
+
+		for (char c : line.toCharArray()) {
+			if (c == '"') {
+				inQuotes = !inQuotes; // toggle state
+			} else if (c == ',' && !inQuotes) {
+				result.add(currentField.toString());
+				currentField.setLength(0); // reset the buffer
+			} else {
+				currentField.append(c);
+			}
+		}
+		result.add(currentField.toString()); // add the last field
+		return result;
+	}
```

Comment:
Is there are strong reason not to use a CSV parsing library? This implementation restricts the CSV formats that can be used and will struggle with some values (escaped quotes).